src/geometry/GeometryEvaluator.cc

```diff
@@ -118,7 +115,8 @@ GeometryEvaluator::ResultObject GeometryEvaluator::applyToChildren(const Abstrac
   for (const auto& item : this->visitedchildren[node.index()]) {
     if (!isValidDim(item, dim)) break;
   }
-  if (dim == 2) return {applyToChildren2D(node, op)};
+  // FIXME: It's annoying to have to disambiguate here:
+  if (dim == 2) return {std::shared_ptr<Geometry>(applyToChildren2D(node, op))};
```

Comment:
ideally we should remove this kind of casts, but we probably need some cursed metaprogramming stuff to do that...

Reply:
I think this is a different issue, originating our combined const/non-const ResultObject type. Ignoring this for now..